teryt/utils_zip.py

```diff
@@ -0,0 +1,80 @@
+import os.path
+import io
+import zipfile
+from collections import OrderedDict
+
+from django.core.management.base import CommandError
+from django.db import transaction, DatabaseError, IntegrityError
+
+import requests
+
+from .utils import get_xml_id_dictionary, parse
+from .models import (RodzajMiejscowosci, JednostkaAdministracyjna,
+                    Miejscowosc, Ulica)
+
+fn_dict = OrderedDict([
+            ('WMRODZ.xml', RodzajMiejscowosci),
+            ('TERC.xml', JednostkaAdministracyjna),
+            ('SIMC.xml', Miejscowosc),
+            ('ULIC.xml', Ulica),
+        ])
+
+url_tmpl = 'http://www.stat.gov.pl/broker/access/'\
+            'prefile/downloadPreFile.jspa?id={}'
+
+
+def open_zipfile_from_url(filename, url):
+    request = requests.get(url, stream=True)
+    zfile = zipfile.ZipFile(io.BytesIO(request.content))
+    return zfile
+
+
+def get_zip_files():
+    zip_files = []
+    xml_dictionary = get_xml_id_dictionary()
+    for file in fn_dict.keys():
+        zfile = open_zipfile_from_url(
+                file,
+                url_tmpl.format(xml_dictionary[file])
+            )
+        zip_files.append(zfile)
+
+    return zip_files
+
+
+def update_database(xml_stream, fname, force_flag):
+    try:
+        c = fn_dict[os.path.basename(fname)]
+    except KeyError as e:
+        raise CommandError('Unknown filename: {}'.format(e))
+
+    try:
+        with transaction.atomic():
+            c.objects.all().update(aktywny=False)
+
+            row_list = parse(xml_stream)
+
+            # MySQL doesn't support deferred checking of foreign key
+            # constraints. As a workaround we sort data placing rows
+            # with no a parent row at the begining.
+            if c is Miejscowosc:
+                row_list = sorted(row_list, key=lambda x: '0000000'
+                                  if x['SYM'] == x['SYMPOD']
+                                  else x['SYM'])
+
+            for vals in row_list:
+                t = c()

```

Comment:
Dobrze jest jak nazwy zmiennych są wiele mówiące. Taka linia na pierwszy rzut oka nie mówi nic. Zamiast c można tą zmienną nazwać np. object_class, a zamiast t np instance. Jak usiądziesz do tego za kilka miesięcy to od razu będziesz wiedział o co chodzi.